Updates to database migrations for foreign keys, theme in Auction.

Ahmed-Elrayes opened the pull request:
- Added Foreign Keys in tables with cascade or set null on update or on delete.

- Added theme:: for AuctionsHouseController.php.
- RankingController.php Uniques section sometimes duplicates the accounts listed when there is multiple pages in paginating and the values weren't accurate so made it to filter the uniques added inside the config/uniques.php.

Please Check them.

Devsome commented on app/Http/Controllers/Frontend/RankingController.php:
```diff
@@ -198,9 +198,15 @@ private function mode($mode, $hideRanking, $hideRankingGuild)
+            foreach ($all_uniques as $key => $unique) {
+                $uniques[] = $key;
+            }
             $jobs = UniqueKillLog::whereNotIn('CharName16', $hideRanking)
```
Why are we searching with `whereIn` in that `dbo.uniquekilllogs`? @Ahmed-Elrayes

Ahmed-Elrayes replied:
yesterday when i checked the unique ranking after i made the pagination to 15, there was 3 pages and on each page there would be few names from the last page and got values as 0, which means it got the uniques which isn't added in the config/uniques.
so after i made it like this it got the correct values and only made one page.
but today my friend called me and told me that the unique ranking made more than 1 page again and got the same characters but the values are different from the previous page and not 0, so i'm gonna re check the controller to figure out why this happens.
Check those images:

![Image 001](https://user-images.githubusercontent.com/30704271/97164763-4b878e00-178b-11eb-86fd-75a388a9092f.png)

![Image 002](https://user-images.githubusercontent.com/30704271/97164765-4cb8bb00-178b-11eb-80de-3aa16ace2cc7.png)

![Image 003](https://user-images.githubusercontent.com/30704271/97164768-4d515180-178b-11eb-858d-22a749641cb9.png)

![Image 004](https://user-images.githubusercontent.com/30704271/97164771-4de9e800-178b-11eb-8da8-2513844dbf79.png)

Ahmed-Elrayes reviewed (commented):
- added the theme:: for all backend controllers so the user can create custom theme view
- fixed unique ranking.

Devsome reviewed (approved):
Watched all files, doing the pr later to fix immediately the translation and creating the docs for unique ranking / sidebar.

Ahmed-Elrayes commented:
as for the last commit 
- Added Pvp Ranking as Free pvp or job pvp in Ranking page.
- Added Latest Free PVP latest kills in sidebar.

and the procedure is the following and put it inside (_AddLogChar):
```
--Character kills table recorder (finish) , all credits goes to #HB
IF (@EventID = 20) -- PVP
BEGIN
    IF ( @desc LIKE '%Trader, Neutral, no freebattle team%'    -- Trader
        OR @desc LIKE '%Hunter, Neutral, no freebattle team%'    -- Hunter
        OR @desc LIKE '%Robber, Neutral, no freebattle team%'    -- Thief
        OR @desc like '%no job, Neutral, %no job, Neutral%'    -- Free PVP
    )
    BEGIN
        -- Get killer name
        DECLARE @killername VARCHAR(512) = @desc
        DECLARE @killeriD INT = 0
		SELECT  @killername = REPLACE(@killername, LEFT(@killername, CHARINDEX('(', @killername)),'')
		SELECT  @killername = REPLACE(@killername, RIGHT(@killername, CHARINDEX(')', REVERSE(@killername))),'')
        SELECT @killeriD = CharID FROM [SRO_VT_SHARD].[dbo].[_Char] WHERE CharName16 = @killername
        -- Get job type
        DECLARE @jobString VARCHAR(10) = LTRIM(RTRIM(SUBSTRING (@desc, 5, 7)))
        DECLARE @jobType INT = CASE
            WHEN @jobString LIKE 'Trader' THEN 1
            WHEN @jobString LIKE 'Robber' THEN 2
            WHEN @jobString LIKE 'Hunter' THEN 3
            ELSE 0 END
        -- Delete original log
        DELETE FROM _LogEventChar WHERE CharID = @CharID AND EventID = 20
            AND (strDesc LIKE '%Trader, Neutral, no freebattle team%'
            OR strDesc LIKE '%Hunter, Neutral, no freebattle team%'
            OR strDesc LIKE '%Robber, Neutral, no freebattle team%'
            OR @desc like '%no job, Neutral, %no job, Neutral%')
        -- Get additional info for notice message
        DECLARE @jobDesc VARCHAR(32) = CASE WHEN @jobType BETWEEN 1 AND 3 THEN 'Job Conflict' ELSE 'Free PVP' END
        DECLARE @strDesc VARCHAR(512)
        IF  (@jobString LIKE 'Trader' OR @jobString LIKE 'Robber' OR @jobString LIKE 'Hunter')
        BEGIN
            -- If it's a Job Kill, then write character nicknames
            DECLARE @killerNickName VARCHAR(64) = (SELECT NickName16 FROM [SRO_VT_SHARD].[dbo].[_Char] WHERE CharID = @killeriD)
            DECLARE @CharnickName VARCHAR(64) = (SELECT NickName16 FROM [SRO_VT_SHARD].[dbo].[_Char] WHERE CharID = @CharID)
            SET @strDesc = '[' + @killerNickName + '] has just killed [' + @CharnickName + '] in [' + @jobDesc + '] mode on [' + CONVERT(NVARCHAR(30), GETDATE(), 0) + ']'
        END

        ELSE BEGIN
            -- If it's normal PVP Kill, write real character names
            SET @strDesc = '[' + @killername + '] has just killed [' + @CharName + '] in [' + @jobDesc + '] mode on [' + CONVERT(NVARCHAR(30), GETDATE(), 0) + ']'
        END
        -- Update the log
        INSERT INTO pvp_records VALUES (@killername, @killeriD, @CharName, @CharID, @jobType, @strPos, @strDesc, GETDATE())
    END
END
```
